jax_distribution/systems/ff_ippo_rware.py

```diff
@@ -0,0 +1,407 @@
+
+    action_dim: Sequence[int]
+    activation: str = "tanh"
+
+    @nn.compact
+    def __call__(
+        self, observation: Observation
+    ) -> Tuple[distrax.Categorical, chex.Array]:
+        """Forward pass."""
+        x = observation.agents_view
+
+        if self.activation == "relu":
+            activation = nn.relu
+        else:
+            activation = nn.tanh
+
+        actor_output = nn.Dense(
+            64, kernel_init=orthogonal(np.sqrt(2)), bias_init=constant(0.0)
+        )(x)
+        actor_output = activation(actor_output)
+        actor_output = nn.Dense(
+            64, kernel_init=orthogonal(np.sqrt(2)), bias_init=constant(0.0)
+        )(actor_output)
+        actor_output = activation(actor_output)
+        actor_output = nn.Dense(
+            self.action_dim, kernel_init=orthogonal(0.01), bias_init=constant(0.0)
+        )(actor_output)
+
+        masked_logits = jnp.where(
+            observation.action_mask,
+            actor_output,
+            jnp.finfo(jnp.float32).min,
+        )
+        actor_policy = distrax.Categorical(logits=masked_logits)
+
+        critic_output = nn.Dense(
+            64, kernel_init=orthogonal(np.sqrt(2)), bias_init=constant(0.0)
+        )(x)
+        critic_output = activation(critic_output)
+        critic_output = nn.Dense(
+            64, kernel_init=orthogonal(np.sqrt(2)), bias_init=constant(0.0)
+        )(critic_output)
+        critic_output = activation(critic_output)
+        critic_output = nn.Dense(
+            1, kernel_init=orthogonal(1.0), bias_init=constant(0.0)
+        )(critic_output)
+
+        return actor_policy, jnp.squeeze(critic_output, axis=-1)
+
+
+def get_learner_fn(
+    env: jumanji.Environment, apply_fn: Callable, update_fn: Callable, config: dict
+) -> Callable:
+    """Get the learner function."""
+
+    def _update_step(runner_state: Tuple, unused_target: Any) -> Tuple:
+        """Update the network."""
+
+        def _env_step(runner_state: Tuple, unused: Any) -> Tuple:
+            """Step the environment."""
+            params, opt_state, rng, env_state, last_timestep = runner_state
+
+            # SELECT ACTION
+            rng, _rng = jax.random.split(rng)
+            actor_policy, value = apply_fn(params, last_timestep.observation)
+            action = actor_policy.sample(seed=_rng)
+            log_prob = actor_policy.log_prob(action)
+
+            # STEP ENVIRONMENT
+            env_state, timestep = jax.vmap(env.step, in_axes=(0, 0))(env_state, action)
+
+            # LOG EPISODE METRICS
+            done, reward = jax.tree_util.tree_map(
+                lambda x: jnp.repeat(x, config["NUM_AGENTS"]).reshape(
+                    config["NUM_ENVS"], -1
+                ),
+                (timestep.last(), timestep.reward),
+            )
+            info = {
+                "returned_episode_returns": env_state.returned_episode_returns,
+                "returned_episode_lengths": env_state.returned_episode_lengths,
```

Comment:
Unclear what this is from the name, can you clarify what this is so we can come up with a better name